src/coupled_driver.cpp

```diff
@@ -329,34 +445,42 @@ void CoupledDriver::init_temperatures()
 {
   comm_.message("Initializing temperatures");
 
-  if (this->has_global_coupling_data()) {
-    auto n_global = this->get_heat_driver().n_global_elem();
-    temperatures_.resize({n_global});
-    temperatures_prev_.resize({n_global});
+  const auto& neutronics = this->get_neutronics_driver();
+  const auto& heat = this->get_heat_driver();
 
-    if (temperature_ic_ == Initial::neutronics) {
+  if (comm_.rank == coupling_root_) {
+    temperatures_.resize({static_cast<unsigned long>(n_global_elem_)});
+    temperatures_prev_.resize({static_cast<unsigned long>(n_global_elem_)});
+  }
+  else if (comm_.rank == neutronics_root_) {
```

Comment:
Looks like the style is off -- can you run clang-format on this?

Reply:
yep!  done.